src/doctor.ts

```diff
@@ -0,0 +1,66 @@
+import { getProtocolInterface } from "./deps.ts";
+
+type RuntimeVersion = {
+  name: string;
+  current: string;
+  minimum?: string;
+  error?: {
+    message: string;
+  };
+};
+
+const getHostedDenoRuntimeVersion = async (): Promise<{
+  minimum?: string;
+  message?: string;
+  error?: { message: string };
+}> => {
+  try {
+    const metadataURL = "https://api.slack.com/slackcli/metadata.json";
```

Comment:
This is a great point! In general, how does this hook work with the CLI's `--slackdev` flag? We may need to duplicate handling the `--slackdev` flag handling that for example the `start` hook is passed (see [here](https://github.com/slackapi/deno-slack-hooks/blob/main/src/mod.ts#L2)).

Reply:
Figured out how to connect the both the `--slackdev` and `--apihost` flags here to replace `api.slack.com` with `api.dev.slack.com` but I'm noticing a few issues that make me hesitant to try powering through...

The `devDomain` used by the `start` hook has authorization support with it from what I can tell. This `devDomain` is used when making SDK requests with some authorized token in the context of a function.

But the https://api.dev.slack.com/slackcli/metadata.json URL is called directly in this hook and behind some authorization. I'm not immediately sure how we can authorize these requests from the hook without including much more information from the CLI 🤔

I also noticed the `check-update` hook uses this same URL, so I'm thinking more exploration into this would be better as a follow up?

Reply:
Saving this work on [the `feat-doctor-hook-dev-domain` branch](https://github.com/slackapi/deno-slack-hooks/compare/feat-doctor-hook...feat-doctor-hook-dev-domain?expand=1#diff-1f70118ca54082607e282e2def0725ca39c7b03913dca09d39660eed955f0068R21) but will move onto handling HTTP errors a bit better for now. No hard feelings if that branch needs to be deleted either!

Reply:
Makes sense! You bring up a good point re: authorization. I think it is worth revisiting in the future, esp. since now we have a few hooks that may need to differentiate between dev and prod, with or without authorization. There's probably a design lurking around here somewhere fits nicely.

Reply:
Good point, the possible implications of authorization is out of scope for the PR